src/coreComponents/physicsSolvers/multiphysics/CoupledReservoirAndWellsBase.hpp

```diff
@@ -175,16 +180,54 @@ class CoupledReservoirAndWellsBase : public CoupledSolver< RESERVOIR_SOLVER, WEL
   WELL_SOLVER *
   wellSolver() const { return std::get< toUnderlying( SolverType::Well ) >( m_solvers ); }
 
-protected:
-
   virtual void
   initializePostInitialConditionsPreSubGroups() override
   {
     Base::initializePostInitialConditionsPreSubGroups( );
 
-    coupledReservoirAndWellsInternal::initializePostInitialConditionsPreSubGroups( this );
+    DomainPartition & domain = this->template getGroupByPath< DomainPartition >( "/Problem/domain" );
```

Comment:
🙈

Reply:
Yes, this is a problem I am having constantly in these initialization functions. I am not aware of a getter for the domain partition that could be used in the physics solvers, but maybe there is one? Or we could add one? That would definitely be cleaner that this.